File: kotlin+java/src/main/kotlin+java/_그리디/뒤집기.java
package _그리디;

import java.util.Scanner;
import java.util.Stack;

public class 뒤집기 {
    public static void main(String [] args) {
        Scanner sc = new Scanner(System.in);
        Stack<Character> stack = new Stack<>();

        String e = sc.nextLine();
        for(int i = 0; i < e.length(); i++){
            if (stack.empty()){
                stack.add(e.charAt(0));
            } else {
                if (stack.lastElement() != e.charAt(i)){
                    stack.add(e.charAt(i));
                }
            }
        }
        int zeroCount = 0;
        for (char ch :stack) {
            if(ch == '0'){
                zeroCount+=1;
            }
        }
        System.out.println(Math.min(stack.size() - zeroCount,zeroCount));
    }
}

File: kotlin+java/src/main/kotlin+java/_그리디/뒤집기.kt
package _그리디

import java.util.Stack

fun main() {
    val input = readlnOrNull() ?: return
    val stack = Stack<Char>()
    input.forEach {
        if (stack.size == 0){
            stack.add(input[0])
        }else{
            if (stack.lastElement() != it){
                stack.add(it)
            }
        }
    }
    val zeroCount = stack.count{ it == '0' }
    println(minOf(zeroCount,stack.size - zeroCount))
}
